src/drivers/CgreenDriver.cpp

```diff
@@ -0,0 +1,170 @@
+#include <cucumber-cpp/internal/drivers/CgreenDriver.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+using namespace cgreen;
+
+namespace cucumber {
+namespace internal {
+
+
+bool CgreenStep::initialized = false;
+
+boost::function< void() > currentTestBody;
+
+static const char* const blacklist[] = {
+        "Running",
+        "Completed",
+        "Failure",
+        "currentTest",
+        "\n",
+        "/"
+};
+
+Ensure(currentTest) {
+    if(currentTestBody)
+    {
+        currentTestBody();
+    }
+}
+
+class CukeCgreenInterceptor {
+public:
+    const std::string getCgreenOutput();
+    static void clearCgreenOutput();
+    static void reset_reporter();
+    static int cgreenPrinter(const char* format, ...);
+    static TestSuite *cgreenSuite;
+    static TestReporter *cgreenReporter;
+    ~CukeCgreenInterceptor();
+
+private:
+    static std::string cgreenOutput;
+    static bool is_blacklisted(const char* input);
+};
+
+static CukeCgreenInterceptor cgreenInterceptor;
+
+std::string CukeCgreenInterceptor::cgreenOutput = std::string();
```

Comment:
Please remove the copy-initialization. In C++ all objects of types that have a default constructor (which is almost all non-POD types) _will_ get it called when no initializer is specified (so no need to worry about uninitialized memory here).